rust/src/pgsql/log.rs

```diff
@@ -0,0 +1,420 @@
+        },
+        PgsqlBEMessage::AuthenticationOk(AuthenticationMessage{
+            identifier: _,
+            length: _,
+            auth_type: _,
+            payload,
+        }) |
+        PgsqlBEMessage::AuthenticationKerb5(AuthenticationMessage{
+            identifier: _,
+            length: _,
+            auth_type: _,
+            payload,
+        }) |
+        PgsqlBEMessage::AuthenticationCleartextPassword(AuthenticationMessage{
+            identifier: _,
+            length: _,
+            auth_type: _,
+            payload,
+        }) |
+        PgsqlBEMessage::AuthenticationMD5Password(AuthenticationMessage{
+            identifier: _,
+            length: _,
+            auth_type: _,
+            payload,
+        }) |
+        PgsqlBEMessage::AuthenticationGSS(AuthenticationMessage{
+            identifier: _,
+            length: _,
+            auth_type: _,
+            payload,
+        }) |
+        PgsqlBEMessage::AuthenticationSSPI(AuthenticationMessage{
+            identifier: _,
+            length: _,
+            auth_type: _,
+            payload,
+        }) |
+        PgsqlBEMessage::AuthenticationGSSContinue(AuthenticationMessage{
+            identifier: _,
+            length: _,
+            auth_type: _,
+            payload,
+        }) |
+        PgsqlBEMessage::AuthenticationSASLFinal(AuthenticationMessage{
+            identifier: _,
+            length: _,
+            auth_type: _,
+            payload,
+        }) => {
+            // TODO question - not sure what method to use here, the format printed doesn't look good
+            js.set_string_from_bytes("payload", payload)?;
+        },
+        PgsqlBEMessage::AuthenticationSASL(_) => {
+            js.set_string("authentication_SASL", &res.to_string())?;
+        },
+        PgsqlBEMessage::AuthenticationSASLContinue(_) => {
+            js.set_string("authentication_SASL_continue", &res.to_string())?;
+        },
+        PgsqlBEMessage::ParameterStatus(ParameterStatusMessage{
+            identifier: _,
+            length: _,
+            param,
+        }) => {
+            js.set_string_from_bytes("parameter_name", &param.param_name)?;
+            js.set_string_from_bytes("parameter_value", &param.param_value)?;
+        },
+        PgsqlBEMessage::BackendKeyData(BackendKeyDataMessage{
+            identifier: _,
+            length: _,
+            backend_pid,
+            secret_key,
+        }) => {
+            js.set_uint("process_id", (*backend_pid) as u64)?;
+            js.set_uint("secret_key", (*secret_key) as u64)?;
+        },
+        PgsqlBEMessage::CommandComplete(
+            RegularPacket{
+                identifier: _,
+                length: _,
+                payload,
+            }) =>
+        {
+            // TODO this may result in not so pretty strings. not sure what to do
+            //  example "SELECT 3\u0000"
+            js.set_string_from_bytes("payload", payload)?;
```

Comment:
Right.  Postgres was clearly written by C developers sending NUL terminated strings over the wire.  I wonder if that should be prettied up here? Anyways, minor nit for now.